src/platform/modulesmanager/src/ModulesManager.cpp

```diff
@@ -663,18 +663,33 @@ int ModulesManager::MpiSetDesired(const char* clientName, const char* payload, i
     }
     else
     {
-        std::string payloadString(payload, payloadSizeBytes);
         rapidjson::Document document;
-        document.Parse(payloadString.c_str());
+        document.Parse(payload, payloadSizeBytes);
 
         if (document.HasParseError())
         {
-            OsConfigLogError(ModulesManagerLog::Get(), "MpiSetDesired invalid payload: %s", payloadString.c_str());
+            if (IsFullLoggingEnabled())
+            {
+                OsConfigLogError(ModulesManagerLog::Get(), "MpiSetDesired invalid payload: %.*s", payloadSizeBytes, payload);
+            }
+            else
+            {
+                OsConfigLogError(ModulesManagerLog::Get(), "MpiSetDesired invalid payload");
+            }

```

Comment:
MpiSetDesired and MpiGetReported are called with high frequency. We need to avoid repetitive loggings that can fill the log files. 

Let's keep only the error logging for when full logging is enabled.

Reply:
Changed to only log if full logging is enabled